lib/block_view_lockups.go

```diff
@@ -1416,18 +2212,20 @@ func (bav *UtxoView) _connectCoinLockupTransfer(
 	}
 
 	// Fetch the sender's balance entries.
-	senderLockedBalanceEntry, err := bav.GetLockedBalanceEntryForHODLerPKIDProfilePKIDUnlockTimestampNanoSecs(
-		senderPKID, profilePKID, txMeta.UnlockTimestampNanoSecs)
+	senderLockedBalanceEntry, err :=
+		bav.GetLockedBalanceEntryForHODLerPKIDProfilePKIDUnlockTimestampNanoSecsVestingEndTimestampNanoSecs(
+			senderPKID, profilePKID, txMeta.UnlockTimestampNanoSecs, txMeta.UnlockTimestampNanoSecs)
 	if err != nil {
 		return 0, 0, nil,
 			errors.Wrap(err, "connectCoinLockupTransfer failed to fetch senderLockedBalanceEntry:w")
 	}
 	if senderLockedBalanceEntry == nil || senderLockedBalanceEntry.isDeleted {
 		senderLockedBalanceEntry = &LockedBalanceEntry{
-			HODLerPKID:              senderPKID,
-			ProfilePKID:             profilePKID,
-			UnlockTimestampNanoSecs: txMeta.UnlockTimestampNanoSecs,
-			BalanceBaseUnits:        *uint256.NewInt(),
+			HODLerPKID:                  senderPKID,
+			ProfilePKID:                 profilePKID,
+			UnlockTimestampNanoSecs:     txMeta.UnlockTimestampNanoSecs,
+			VestingEndTimestampNanoSecs: txMeta.UnlockTimestampNanoSecs,
```

Comment:
Nit: I would do this a bit differently. I would validate that txMeta.UnlockTimestampNanoSecs == txMeta.VestingTimestampNanoSecs at the top, and then use the proper values from txMeta here. The way you've done it now makes it very error-prone if we want to make vested lockups transferrable in the future.